### DISCUSSION TITLE
Closed-loop Linkages Support

### DISCUSSION THREAD

**Discussion by Cfather** [isaac-sim/IsaacLab#1453] [Q&A] [answered]
Hi, I am a humanoid developer. Closed-loop linkages are quite common in the recent hardware design of humanoids. I wonder if IsaacLab or PhysX supports the dynamics of robotic systems with closed-loop linkages? Thanks!

**Comment by RandomOakForest** [comment]
Thank you for your interest in IsaacLab. Closed-loop linkages are supported by PhysX, which IsaacLab can leverage. [This doc may help](https://nvidia-omniverse.github.io/PhysX/physx/5.4.2/docs/Articulations.html#closing-loops).

**Reply by shanereetz** [reply]
@Cfather

If you're rigging the design in Isaac Sim or other Omniverse apps, look for the "Exclude from Articulation" flag on joints. This will need to be applied on one joint in the loop. The solver will accumulate more error at that joint.

https://docs.omniverse.nvidia.com/extensions/latest/ext_physics/articulations.html#closing-loops

**Reply by Cfather** [reply]
Thanks for your responses! I wonder if there is a simple example USD file that sets up a simple robotic system with linkages, like a 4-bar linkage.

**Comment by StrainFlow** [comment, marked as answer]
Here is a simple crank-slider you in OnShape that you can use to practice rigging: 

https://cad.onshape.com/documents/d1760add554ea0473707b7a7/w/2a05beb5a9a8e66221e3f4ff/e/f5f07976001089b0b900bf13?renderMode=0&uiState=6750a2135da8c93933bff1bc

You can make your own copy on OnShape and then use the OnShape importer to convert it to USD. 

https://github.com/user-attachments/assets/ccfe39e9-dbba-40a7-8ce2-5594eeb9f881
